app/server/utils/metrics_cache.py

```diff
@@ -0,0 +1,89 @@
+from datetime import datetime, timedelta
+
+from server import red, db
+from flask import g
+from decimal import Decimal
+import json
+import pickle
+
+SUM = 'SUM'
+SUM_OBJECTS ='SUM_OBJECTS'
+TIMESERIES ='TIMESERIES'
+COUNT ='COUNT'
+
+valid_strategies = [SUM, TIMESERIES, COUNT, SUM_OBJECTS]
+
+def _store_cache(key, value):
+    pickled_object = pickle.dumps(value)
+    red.set(key, pickled_object)
+    return True
+
+def _load_cache(key):
+    cached_object = red.get(key)
+    if not cached_object:
+        return None
+    return pickle.loads(cached_object)       
```

Comment:
I got a corrupted cache, and it caused pickle to die. This allowed it to recover:

```
def _load_cache(key):
    cached_object = red.get(key)
    if not cached_object:
        return None
    try:
        return pickle.loads(cached_object)     
    except pickle.UnpicklingError:
        red.delete(key)
        return None
```

Reply:
Good catch. Do you remember the conditions that made that happen? I did make that change though, also had to add the following chunk of code to ensure that a corrupted cache won't lead to undercounted results 
```
    # If there's no cache (either it's a new attribute we're tracking, or the cache is corrupted)
    # then we should pull results starting at id=0
    if cache_result:
        filtered_query = query.filter(object_model.id > highest_id_in_cache)
    else:
        filtered_query = query
```